freqtrade/rpc/api_server/api_auth.py

```diff
@@ -44,6 +48,32 @@ def get_user_from_token(token, secret_key: str, token_type: str = "access"):
     return username
 
 
+# This should be reimplemented to better realign with the existing tools provided
+# by FastAPI regarding API Tokens
+# https://github.com/tiangolo/fastapi/blob/master/fastapi/security/api_key.py
+async def validate_ws_token(
+    ws: WebSocket,
+    ws_token: Union[str, None] = Query(default=None, alias="token"),
+    api_config: Dict[str, Any] = Depends(get_api_config)
+):
+    secret_ws_token = api_config.get('ws_token', 'secret_ws_t0ken.')
+    secret_jwt_key = api_config.get('jwt_secret_key', 'super-secret')
+
+    if ws_token and secrets.compare_digest(secret_ws_token, ws_token):
```

Comment:
we will have to have `secret_ws_token` default to None - and then check for that in the IF explicitly

as is, it's a security problem - assuming all running bots update, 90% will not add a ws_token config initially - so you can go around the internet trying `secret_ws_t0ken.` - you'll surely find one or the other ...

Reply:
Ope, good catch